lxd/networks.go

```diff
@@ -313,23 +352,25 @@ func networksGet(d *Daemon, r *http.Request) response.Response {
 	resultString := []string{}
 	resultMap := []*api.Network{}
 	urlToNetwork := make(map[*api.URL]auth.EntitlementReporter)
-	for kind, networkNames := range networks {
-		for _, networkName := range networkNames {
-			// Filter out managed networks that the caller doesn't have permission to view.
-			if kind == managed && !userHasPermission(entity.NetworkURL(requestProjectName, networkName)) {
-				continue
-			}
-
-			if !recursion {
-				resultString = append(resultString, "/"+version.APIVersion+"/networks/"+networkName)
-			} else {
-				net, err := doNetworkGet(s, r, s.ServerClustered, requestProjectName, reqProject.Config, networkName)
-				if err != nil {
+	for kind, projectNetworks := range networks {
+		for projectName, networkNames := range projectNetworks {
+			for _, networkName := range networkNames {
+				// Filter out managed networks that the caller doesn't have permission to view.
+				if kind == managed && !userHasPermission(entity.NetworkURL(projectName, networkName)) {
 					continue
 				}
 
-				resultMap = append(resultMap, &net)
-				urlToNetwork[entity.NetworkURL(requestProjectName, networkName)] = &net
+				if !recursion {
+					resultString = append(resultString, "/"+version.APIVersion+"/networks/"+networkName)
```

Comment:
here's a good place to use URL builder so that networkName is path escaped automatically